tasks/main.yml

```diff
@@ -6,59 +6,41 @@
       set_fact:
         rsyslog_enabled: "{{ logging_enabled | d(true) }}"
 
-    - name: Set rsyslog_elasticsearch
+    - name: Set rsyslog_output_elasticsearch for the omelasticsearch output
       set_fact:
-        rsyslog_elasticsearch: "{{ rsyslog_elasticsearch | d([]) }} + {{ [ item.0 | combine( {'logs_collections_name': item.1.name} )] }}"
-      with_subelements:
-        - "{{ logging_outputs }}"
-        - logs_collections
+        rsyslog_output_elasticsearch: "{{ rsyslog_output_elasticsearch | d([]) }} + {{ [ item ] }}"
+      loop: "{{ logging_outputs }}"
       when:
-        - item.0.type | d('') == 'elasticsearch'
-        - item.1.state | d('present') == 'present'
+        - item.name is defined
+        - item.type | d('') == 'elasticsearch'
 
-    - name: Set rsyslog_files_actions
+    - name: Set rsyslog_output_files for the omfile outputs
       set_fact:
-        rsyslog_files_actions: "{{ rsyslog_files_actions | d([]) }} + {{ item.0.rsyslog_files_actions }}"
-      with_subelements:
-        - "{{ logging_outputs }}"
-        - logs_collections
+        rsyslog_output_files: "{{ rsyslog_output_files | d([]) }} + {{ [ item ] }}"
+      loop: "{{ logging_outputs }}"
       when:
-        - item.0.type | d('') == 'files'
-        - item.0.rsyslog_files_actions is defined
-        - item.1.state | d('present') == 'present'
+        - item.name is defined
+        - item.type | d('') == 'files'
 
-    - name: Set rsyslog_forwards_actions
+    - name: Set rsyslog_output_forwards for the omfwd output
       set_fact:
-        rsyslog_forwards_actions: "{{ rsyslog_forwards_actions | d([]) }} + {{ item.0.rsyslog_forwards_actions }}"
-      with_subelements:
-        - "{{ logging_outputs }}"
-        - logs_collections
+        rsyslog_output_forwards: "{{ rsyslog_output_forwards | d([]) }} + {{ [ item ] }}"
+      loop: "{{ logging_outputs }}"
       when:
-        - item.0.type | d('') == 'forwards'
-        - item.0.rsyslog_forwards_actions is defined
-        - item.1.state | d('present') == 'present'
-
-    - name: Set rsyslog_logs_collections
-      set_fact:
-        rsyslog_logs_collections: "{{ rsyslog_logs_collections | d([]) }} + {{ [ item.1 | combine( {'output_name': item.0.name} )] }}"
-      with_subelements:
-        - "{{ logging_outputs }}"
-        - logs_collections
-
-    - name: Initialize rsyslog_outputs
-      set_fact:
-        rsyslog_outputs: []
+        - item.name is defined
+        - item.target is defined
+        - item.type | d('') == 'forwards'
 
     - name: Set rsyslog_outputs
       set_fact:
-        rsyslog_outputs: "{{ rsyslog_outputs | d([]) }} + {{ [ item.type ] }}"
-      with_items: "{{ logging_outputs }}"
-      when: item.type is defined
+        rsyslog_outputs: '{{ rsyslog_output_elasticsearch | d([]) }} + {{ rsyslog_output_files | d([]) }} + {{ rsyslog_output_forwards | d([]) }}'
```

Comment:
is `rsyslog_outputs` the same as `logging_outputs`?

Reply:
Basically, yes.  I expect it's a cleaned up version of logging_outputs. :)

Reply:
Then can you just do 
```
rsyslog_outputs: "{{ logging_outputs }}"
```
?  I guess I don't understand what you mean by "cleaned up" here - is it that `rsyslog_output_elasticsearch | d([]) }} + {{ rsyslog_output_files | d([]) }} + {{ rsyslog_output_forwards | d([])` is somehow different than `logging_outputs`?

Reply:
For instance, this removes an elasiticsearch type with no name parameter, doesn't it?
`rsyslog_output_elasticsearch: "{{ logging_outputs | selectattr('name','defined') | selectattr('type','defined') | selectattr('type','==','elasticsearch') | list }}"`

Reply:
Yes, it does.

Under what conditions is `logging_outputs` not the same as `rsyslog_output_elasticsearch + rsyslog_output_files + rsyslog_output_forwards`?  Is it when the user specifies some bogus values in `logging_outputs`?  Is it when the user specifies some outputs that are not yet supported by the `rsyslog` role?

Reply:
> Under what conditions is `logging_outputs` not the same as `rsyslog_output_elasticsearch + rsyslog_output_files + rsyslog_output_forwards`? Is it when the user specifies some bogus values in `logging_outputs`? 

I think bogus values would go through.  It just removes `name` is not defined.

> Is it when the user specifies some outputs that are not yet supported by the `rsyslog` role?

Yes, they will be dropped.